services/wfm/src/main/java/org/openkilda/wfm/share/service/SpeakerFlowSegmentRequestBuilder.java

```diff
@@ -80,15 +89,18 @@ public SpeakerFlowSegmentRequestBuilder(FlowResourcesManager resourcesManager) {
     @Override
     public List<FlowSegmentRequestFactory> buildIngressOnly(
             CommandContext context, Flow flow, FlowPath path, FlowPath oppositePath) {
-        return makeRequests(context, flow, path, oppositePath, true, false, false);
+        return makeRequests(context, flow, path, oppositePath, true, false, false,
+                SpeakerRequestBuildContext.builder().build());
     }
 
     private List<FlowSegmentRequestFactory> makeRequests(
             CommandContext context, Flow flow, FlowPath path, FlowPath oppositePath,
-            boolean doIngress, boolean doTransit, boolean doEgress) {
+            boolean doIngress, boolean doTransit, boolean doEgress,
+            SpeakerRequestBuildContext speakerRequestBuildContext) {
         if (path == null) {
             path = oppositePath;
             oppositePath = null;
+            speakerRequestBuildContext.swapMainAndOppositeData();
```

Comment:
I'd prefer another instance returned by swapMainAndOppositeData and used within the method, instead of modifying object that comes as an input argument.

Reply:
Fixed.